src/main/java/com/aws/greengrass/detector/config/Config.java

```diff
@@ -30,27 +32,22 @@
      */
     public Config(Topics topics) {
         Topics configurationTopics = topics.lookupTopics(KernelConfigResolver.CONFIGURATION_CONFIG_KEY);
-        configurationTopics.subscribe((whatHappened, node) -> {
-            // Hardcoding port for now till MQTT Broker is not publishing it.
-            this.mqttPort = new AtomicInteger(DEFAULT_MQTT_PORT);
+        if (configurationTopics != null) {
+            configurationTopics.subscribe((whatHappened, node) -> {
+                if (configurationTopics.isEmpty()) {
+                    this.includeIPv4LoopbackAddrs.set(DEFAULT_INCLUDE_IPV4_LOOPBACK_ADDRESSES);
+                    this.includeIPv4LinkLocalAddrs.set(DEFAULT_INCLUDE_IPV4_LINK_LOCAL_ADDRESSES);
+                    return;
+                }
 
-            if (configurationTopics.isEmpty()) {
-                this.includeIPv4LoopbackAddrs = new AtomicBoolean(DEFAULT_INCLUDE_IPV4_LOOPBACK_ADDRESSES);
-                this.includeIPv4LinkLocalAddrs = new AtomicBoolean(DEFAULT_INCLUDE_IPV4_LINK_LOCAL_ADDRESSES);
-                return;
-            }
-
-            this.includeIPv4LoopbackAddrs = new AtomicBoolean(
-                    Coerce.toBoolean(
-                            configurationTopics.findOrDefault(
-                                    DEFAULT_INCLUDE_IPV4_LOOPBACK_ADDRESSES,
-                                    INCLUDE_IPV4_LOOPBACK_ADDRESSES_CONFIG_KEY)));
-            this.includeIPv4LinkLocalAddrs = new AtomicBoolean(
-                    Coerce.toBoolean(
-                            configurationTopics.findOrDefault(
-                                    DEFAULT_INCLUDE_IPV4_LINK_LOCAL_ADDRESSES,
-                                    INCLUDE_IPV4_LINK_LOCAL_ADDRESSES_CONFIG_KEY)));
-        });
+                this.includeIPv4LoopbackAddrs.set(Coerce.toBoolean(configurationTopics
+                        .findOrDefault(DEFAULT_INCLUDE_IPV4_LOOPBACK_ADDRESSES,
+                                INCLUDE_IPV4_LOOPBACK_ADDRESSES_CONFIG_KEY)));
+                this.includeIPv4LinkLocalAddrs.set(Coerce.toBoolean(configurationTopics
+                        .findOrDefault(DEFAULT_INCLUDE_IPV4_LINK_LOCAL_ADDRESSES,
+                                INCLUDE_IPV4_LINK_LOCAL_ADDRESSES_CONFIG_KEY)));
+            });
```

Comment:
Can we add a log here for updated config?

Reply:
Done